js/data/utils.js

```diff
@@ -226,6 +231,143 @@ var isUnaryOperation = function(crit) {
     return crit[0] === "!" && Array.isArray(crit[1]);
 };
 
+var trivialPromise = function() {
+    var d = new Deferred();
+    return d.resolve.apply(d, arguments).promise();
+};
+
+var rejectedPromise = function() {
+    var d = new Deferred();
+    return d.reject.apply(d, arguments).promise();
+};
+
+function ArrayHelper() {
```

Comment:
Why is ArrayHelper class needs? It does not contain any state.  Why methods push, updateArrayItem insertItemToArray, removeItemFromArray are not exported?